### PR TITLE
[#211] Option to specify servers in Azure domain mode

### LINKED ISSUE #211: FR specify server-group for domain deployment
Currently sunstone tries to deploy to all server-groups available, this may prove problematic when groups have different profiles . Deployment on all groups fails when one of the groups is missing services/dependencies. My current workaround is to destroy all unused server-groups.

### PR DESCRIPTION
**Pull request by laDok8**
this PR closes #211

### CODE REVIEW

File: azure-wildfly/src/main/java/sunstone/azure/impl/AzureWFArchiveDeployer.java [hunk 1]
```diff
@@ -88,11 +94,32 @@ static void undeployFromWebApp(Identification resourceIdentification, AzureSunst
     }
 
     static void deployToVmInstance(String deploymentName, Identification resourceIdentification, WildFly wildFly, InputStream is, AzureSunstoneStore store) throws SunstoneException {
+        Deploy.Builder builder = new Deploy.Builder(is, deploymentName, false);
```
Comment (istraka) [inline azure-wildfly/src/main/java/sunstone/azure/impl/AzureWFArchiveDeployer.java:97]: This seems to be simmilar to what is in the `aws-wildfly` module. Wouldn't it be worth it to have a functionality in a util class in `core-wildfly` module?

**Review by istraka — CHANGES_REQUESTED** [review]
Do we need cutomImage template? Wouldn't it be worth it to change both standalone and domain template to use marketplace image?

**Comment by laDok8** [comment]
> Do we need cutomImage template? Wouldn't it be worth it to change both standalone and domain template to use marketplace image?

I'm not sure about our requirements. wouldn't we lose test coverage this way, it's true that more tests are in clouds-ts but this way error would be caught earlier on @istraka

**Comment by istraka** [comment]
I don't think we would. The goal is to test the Sunstone framework, not EAP. And I don't think we care about custom images in this point.

File: azure-wildfly/src/test/resources/sunstone.properties [hunk 2]
```diff
@@ -13,4 +13,12 @@ sunstone.wildfly.mgmt.connection.timeout=120000
 sunstone.wildfly.mgmt.host=master
 sunstone.wildfly.mgmt.profile=default
 
+# valid for marketplace-image profile
+ts.azure.eap.image.plan.name=${azure.image.plan.name}
```
Comment (istraka) [inline azure-wildfly/src/test/resources/sunstone.properties:17]: Perhaps we can set some defaults here. WDYT?

Comment (laDok8) [reply azure-wildfly/src/test/resources/sunstone.properties:17]: good idea

File: azure-wildfly/src/test/resources/sunstone.properties [hunk 3]
```diff
@@ -13,4 +13,11 @@ sunstone.wildfly.mgmt.connection.timeout=120000
 sunstone.wildfly.mgmt.host=master
 sunstone.wildfly.mgmt.profile=default
 
+ts.azure.eap.image.plan.name=${azure.image.plan.name:rh-jboss-eap74-rhel8}
+ts.azure.eap.image.publisher=${azure.image.publisher:redhat}
+ts.azure.eap.image.product=${azure.image.product:rh-jboss-eap}
+ts.azure.eap.image.offer=${azure.image.offer:rh-jboss-eap-PREVIEW}
```
Comment (istraka) [inline azure-wildfly/src/test/resources/sunstone.properties:19]: Can you put `rh-jboss-eap` there? `preview` is a stage with limited access.

File: azure-wildfly/src/test/resources/sunstone.properties [hunk 4]
```diff
@@ -13,4 +13,11 @@ sunstone.wildfly.mgmt.connection.timeout=120000
...
+ts.azure.eap.image.sku=${azure.image.sku:rh-jboss-eap74-rhel8}
+ts.azure.eap.image.version=${azure.image.version:8.6.2023121117}
```
Comment (istraka) [inline azure-wildfly/src/test/resources/sunstone.properties:21]: IMO `latest` should be here.

File: azure/src/test/resources/sunstone.properties [hunk 5]
```diff
@@ -13,4 +13,11 @@ sunstone.wildfly.mgmt.connection.timeout=120000
 sunstone.wildfly.mgmt.host=master
 sunstone.wildfly.mgmt.profile=default
 
+ts.azure.eap.image.plan.name=${azure.image.plan.name:rh-jboss-eap74-rhel8}
+ts.azure.eap.image.publisher=${azure.image.publisher:redhat}
+ts.azure.eap.image.product=${azure.image.product:rh-jboss-eap}
+ts.azure.eap.image.offer=${azure.image.offer:rh-jboss-eap-PREVIEW}
```
Comment (istraka) [inline azure/src/test/resources/sunstone.properties:19]: Same here and below.

**Review by istraka — CHANGES_REQUESTED** [review]

**Review by istraka — APPROVED** [review]

**Pull request merged**